cray/modules/console/cli.py

```diff
@@ -0,0 +1,203 @@
+#
+#  MIT License
+#
+#  (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+#  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+#  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#  OTHER DEALINGS IN THE SOFTWARE.
+#
+""" Console - websocket interaction with console services """
+# pylint: disable=invalid-name,redefined-outer-name,missing-docstring,unused-argument,broad-except,too-many-locals,too-many-branches,too-many-statements
+import asyncio
+import json
+import click
+import websockets
+import aioconsole
+
+from cray.constants import TENANT_HEADER_NAME_KEY
+from cray.core import argument
+from cray.core import group
+from cray.core import pass_context
+from cray.core import option
+
+# Header name keys for console apis
+CONSOLE_HEADER_TAIL_KEY = "Cray-Console-Lines"
+CONSOLE_HEADER_FOLLOW_KEY = "Cray-Console-Follow"
+
+@group()
+def cli():
+    """ Interact with node consoles """
+    pass
+
+###########################################################################
+# helper function to run the websocket terminal interaction
+###########################################################################
+async def websocket_terminal_interaction(ctx, endpoint: str, headers: dict[str,str]) -> str:
+    # set up the return value
+    errMsg = ""
+
+    # pull information from context
+    config = ctx.obj['config']
+    auth = ctx.obj["auth"]
+    globals_ctx = ctx.obj['globals']
+    tenant = config.get('core.tenant', "")
+    hostname = config.get('core.hostname', "")
+
+    # add auth header
+    token = ""
+    if auth and auth.session and auth.session.access_token:
+        token = auth.session.access_token
+    elif globals_ctx and globals_ctx.token.access_token:
+        token = globals_ctx.token.access_token
```

Comment:
The `token` `AssertionError` issue is being caused by the fact that globals_ctx is a dictionary, and so is globals_ctx["token"] (if it is set). This change fixes it:
```suggestion
    globals_token = ctx.obj.get("globals",{}).get("token",{}).get("access_token","")
    tenant = config.get('core.tenant', "")
    hostname = config.get('core.hostname', "")

    # add auth header
    token = ""
    if auth and auth.session and auth.session.access_token:
        token = auth.session.access_token
    else:
        token = globals_token
```

Reply:
Fixed